File: Machine_Learning/Decision_Tree_Regression.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Wed Feb 15 11:27:14 2017

@author: gaulbg
"""

#Data Preprocessing

#Import Libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Import Dataset
dataset = pd.read_csv("Position_Salaries.csv")
x = dataset.iloc[:,1:2].values
y = dataset.iloc[:,2].values

##Feature Scaling
#from sklearn.preprocessing import StandardScaler
#sc_x = StandardScaler()
#x = sc_x.fit_transform(x)
#sc_y = StandardScaler()
#y = sc_y.fit_transform(y)
               
# Fitting Decision Tree to data set
from sklearn.tree import DecisionTreeRegressor
regressor = DecisionTreeRegressor(random_state = 0)
regressor.fit(x, y)

# Predicting 
y_pred = regressor.predict(6.5)


#Increment precision in X
x_grid = np.arange(min(x), max(x), 0.01)
x_grid = x_grid.reshape((len(x_grid), 1))
plt.scatter(x, y, color = 'green')
plt.plot(x_grid, regressor.predict(x_grid), color = 'pink')
plt.title('Decision Tree')
plt.xlabel('Position Level')
plt.ylabel('Salary')

File: Machine_Learning/Decision_Tree_Regression.R
#Decision Tree Regression

#install.packages('ggplot2')

#Data Processing
setwd("~/Git/Udemy/Machine_Learning")
dataset = read.csv("Position_Salaries.csv")
dataset = dataset[2:3]

# #Feature Scaling
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])
# 

# Fitting Decision Tree Regression
# install.packages('rpart')
library(rpart)
regressor = rpart(formula = Salary ~ .,
                  data = dataset,
                  control = rpart.control(minsplit = 1))

# Predict with Decision Tree Regression
y_pred = predict(regressor, newdata = data.frame(Level = 6.5))

# Visualizing Linear model
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
ggplot() + 
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             color = 'red') + 
  geom_line(aes(x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))),
            color = 'blue') +
  ggtitle('Decision Tree Regression') + 
  xlab('Leavel') + 
  ylab('Salary')

